UI/src/utils/backend-utils.js

```diff
@@ -665,26 +665,26 @@
     }
 
     function latestResponseRequestor(getParameters) {
+      var deferred;
+      var request;
 
-        var deferred;
-        var request;
+      function doRequest() {
+        eventbus.trigger('loadingBar:show');
+        if (request)
+          request.abort();
 
-        function doRequest(){
-
-            if(request)
-                request.abort();
-
-            request = $.ajax(getParameters.apply(undefined, arguments)).done(function(result){
-                deferred.resolve(result);
-            });
-            return deferred;
-        }
+        request = $.ajax(getParameters.apply(undefined, arguments)).done(function (result) {
+          deferred.resolve(result);
+          eventbus.trigger('loadingBar:hide');
```

Comment:
Should we hide the loading bar when the request is finished or when the assets are rendered on the map? If there are a lot of assets there can probably be some second of delay between the request and the rendering

Reply:
> The only problem I see is the paralel calls like the manouvres/mass transit stops (these types of special cases) since one call will terminate and the other will not but will remove the loading bar nonetheless on the first call teminated
Fixed this one problem